core/commands/publish.go

```diff
@@ -96,7 +97,18 @@ Publish an <ipfs-path> to another public key (not implemented):
 			popts.pubValidTime = d
 		}
 
-		output, err := publish(req.Context(), n, n.PrivateKey, path.Path(pstr), popts)
+		ctx := req.Context()
+		if ttl, found, _ := req.Option("ttl").String(); found {
+			d, err := time.ParseDuration(ttl)
+			if err != nil {
+				res.SetError(err, cmds.ErrNormal)
+				return
+			}
+
+			ctx = context.WithValue(ctx, "ipns-publish-ttl", d)
```

Comment:
:+1: we should enhance the context much more often.

This would allow us to get a trace of where it failed in timeouts instead of a blank "deadline exceeded".